servers/src/main/java/tachyon/worker/file/FileDataManager.java

```diff
@@ -75,13 +77,21 @@ public synchronized void persistFile(long fileId, List<Long> blockIds) throws IO
     OutputStream outputStream = mUfs.create(dstPath);
     final WritableByteChannel outputChannel = Channels.newChannel(outputStream);
 
+    Map<Long, Long> blockIdToLockId = Maps.newHashMap();
+    // lock all the blocks to prevent any eviction
     for (long blockId : blockIds) {
       long lockId;
       try {
         lockId = mBlockDataManager.lockBlock(Sessions.CHECKPOINT_SESSION_ID, blockId);
+        blockIdToLockId.put(blockId, lockId);
       } catch (BlockDoesNotExistException e) {
         throw new IOException(e);
```

Comment:
Thanks. But it's tricky here that the unlock can also throw exceptions. So to ensure all the block ids can be released, we have to buffer the exceptions.